app/web/modules/stripe/stripe.model.js

```diff
@@ -1,87 +1,53 @@
 const Joi = require('joi')
+const _ = require('lodash')
 
 const { ValidationError } = require('./../../../lib/errors')
+const { addEntityIfValid, stripEmpty } = require('./../../../lib/validation')
+
+const StripeBankAccount = require('./bankAccount.model')
+const StripeLegalEntity = require('./legalEntity.model')
 
 const schema = {
-  bank_account_sort_code: Joi.string().required(),
-  bank_account_number: Joi.string().required(),
-  org_address_line_1: Joi.string().required(),
-  org_address_city: Joi.string().required(),
-  org_address_postcode: Joi.string().required(),
-  person_dob_day: Joi.number().required(),
-  person_dob_month: Joi.number().required(),
-  person_dob_year: Joi.number().required(),
   org_id: Joi.string().required(),
   org_name: Joi.string().required(),
-  person_address_line_1: Joi.string().required(),
-  person_address_city: Joi.string().required(),
-  person_address_postcode: Joi.string().required(),
-  person_first_name: Joi.string().required(),
-  person_last_name: Joi.string().required(),
   org_ip_address: Joi.string().required(),
   org_statement_descriptor: Joi.string().required(),
   org_phone_number: Joi.string().required()
 }
 
 class StripeAccount {
   constructor (body) {
-    const { error, value: model } = Joi.validate(body, schema, { allowUnknown: true, stripUnknown: true })
+    const { error, value: model } = Joi.validate(body, schema, { allowUnknown: true })
 
     if (error) {
       throw new ValidationError(`StripeAccount ${error.details[0].message}`)
     }
 
-    Object.assign(this, this.applyDefaults(model))
+    Object.assign(this, this.build(model))
   }
 
   basicObject () {
     return Object.assign({}, this)
   }
 
-  applyDefaults (params) {
-    const defaults = {
+  build (params) {
+    let core = {
```

Comment:
It looks like this can be a `const`, not 100% if that would cause issues with the `addEntityIfValid` method but it should be alright.

Reply:
good shout